Created Club Profile Page

laurasofi1024 opened the pull request:
Established Club Profile Page

Created Component for the Main Content of the Profile Page

laurasofi1024 commented:
Current Desktop View:
![Screen Shot 2020-11-13 at 4 23 51 PM](https://user-images.githubusercontent.com/61126997/99122677-c85c9980-25cc-11eb-98ce-2a7cb134df8d.png)

laurasofi1024 commented:
@erin2722  I made the application link a button. This was improvised so please let me know if I should change it back to a link or style it differently

erin2722 reviewed (changes requested):
This looks incredible!

I think that looking at Francesca's explore page branch will help clarify a lot of the questions you sent, and I hope to get that merged in quite soon. The basic infrastructure that you have commented out is exactly right. 🎉 👏 

For now, maybe add in the meeting frequency box, the extra button, and the tablet view, and then once francesca's PR is merged in we will get this working even better 🥇

laurasofi1024 commented:
Changes made:
- Added Join mailing list button
- Created Tablet View
- Changed location of apply button per the updated Figma mockup
- Made changes to margins of FrequencyTag and ProfilePageBox components

General comments:
- I assume there will be a specific link for the mailing list added to the database later on so I just made its button's link `club.mailingList` for now
- The FrequencyTag component requires a string for frequency (i.e "1", "2", "3" etc) and a boolean for weekly. I put `club.meetingFrequency` and `club.weekly` for each of these respectively for now pending how these will be represented in the database. 
- How can I access the database for my local app? I assume I have to download MongoDB locally or something along those lines

laurasofi1024 commented:
Current Desktop View:
![Screen Shot 2020-11-28 at 10 51 41 PM](https://user-images.githubusercontent.com/61126997/100533232-07dfd400-31d0-11eb-9e1e-4204540b660f.png)

Current Mobile View:
![localhost_3000_clubs(iPhone X)](https://user-images.githubusercontent.com/61126997/100533235-0adac480-31d0-11eb-8f0a-52b3912a6e08.png)

Current Tablet View:
![localhost_3000_clubs(iPad)](https://user-images.githubusercontent.com/61126997/100533236-0f06e200-31d0-11eb-840f-2f5ed70c5983.png)

erin2722 commented on client/src/components/profileMainContent/index.js:
```diff
@@ -0,0 +1,57 @@
+            <h2>Description</h2>
+            <p className='one'>{description}</p>
+            <h2>Highlights</h2>
+            <p className='two'>{highlights.map(highlight => <Highlight text={highlight}/>)}</p>
```
this looks great!

erin2722 commented on client/src/containers/ClubProfile.js:
```diff
@@ -0,0 +1,264 @@
+          })
+          .catch((error) => console.log(error));
+    }, [id]);
+    
```
I would recommend adding this variable and then replacing all instances of the similar clubs code with `<SimilarClubs />` in order to make the code shorter/more readable
```suggestion
        const SimilarClubs = () => { return (club.similarClubs.length === 0) ? (<h1>Loading</h1>) : 
        (club.similarClubs.map(profile => (
            <ExploreBox 
                name = {profile.name}
                description = {profile.shortDescription}
                imageURL = {profile.imageUrl}
                tags = {profile.tags}
                clubSize = {profile.memberRange}
                acceptingMembers = {profile.acceptingMembers}
                applicationRequired = {profile.applicationRequired}
                cardLink={`/clubs/${profile._id}`}
            />
    )))};
```

erin2722 commented on client/src/containers/ClubProfile.js:
```diff
@@ -0,0 +1,264 @@
+                                    howToJoin= {club.howToJoin}
+                                />
+                                <h2>Similar Clubs</h2>
+                                {(club.similarClubs.length === 0) ? (<h1>Loading</h1>) : (club.similarClubs.map(profile => (
+                                    <ExploreBox 
+                                        name = {profile.name}
+                                        description = {profile.shortDescription}
+                                        imageURL = {profile.imageUrl}
+                                        tags = {profile.tags}
+                                        clubSize = {profile.memberRange}
+                                        acceptingMembers = {profile.acceptingMembers}
+                                        applicationRequired = {profile.applicationRequired}
+                                        cardLink={`/clubs/${profile._id}`}
+                                    />
+                                )))}    
```
```suggestion
<SimilarClubs />
```

erin2722 reviewed (changes requested):
🎉 this looks incredible! 

After the small changes we will get this merged in!
